service/matching/pri_matcher.go

```diff
@@ -403,9 +423,13 @@ again:
 		}
 		return nil, res.ctxErr
 	}
-	bugIf(res.poller == nil, "bug: bad match result in syncOfferTask")
+	if !softassert.That(tm.logger, res.poller != nil, "expected poller from match") {
+		return nil, errors.New("internal error: no poller found on sync match")
```

Comment:
should we just define a `errInternalMatchError = serviceerror.NewInternal("internal matcher error")` and use that?
just a thought, either way is fine